CMakeLists.txt

```
@@ -0,0 +1,186 @@
+    # ui
+    ${PROJECT_SOURCE_DIR}/src/aboutwindow.ui
+    ${PROJECT_SOURCE_DIR}/src/mainwindow.ui
+    ${PROJECT_SOURCE_DIR}/src/styleeditorwindow.ui
+    ${PROJECT_SOURCE_DIR}/src/updaterwindow.ui
+    # 3rdParty ui
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qplaintexteditsearchwidget.ui
+    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/src/Downloader.ui
+    # 3rdParty source files
+    ${PROJECT_SOURCE_DIR}/3rdParty/qautostart/src/qautostart.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/qautostart/src/qautostart.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/markdownhighlighter.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qmarkdowntextedit.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qownlanguagedata.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qplaintexteditsearchwidget.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/src/Downloader.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/src/QSimpleUpdater.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/src/Updater.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/qxt/qxtglobal.cpp
+    ${PROJECT_SOURCE_DIR}/3rdParty/qxt/qxtglobalshortcut.cpp
+    # qrc files
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/media.qrc
+    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/etc/resources/qsimpleupdater.qrc
+    ${PROJECT_SOURCE_DIR}/src/fonts.qrc
+    ${PROJECT_SOURCE_DIR}/src/images.qrc
+    ${PROJECT_SOURCE_DIR}/src/styles.qrc
+)
+
+file (GLOB_RECURSE HEADERS
+    ${PROJECT_SOURCE_DIR}/3rdParty/qautostart/qautostart.h
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/linenumberarea.h
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/markdownhighlighter.h
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qmarkdowntextedit.h
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qownlanguagedata.h
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qplaintexteditsearchwidget.h
+    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/include/*.h
+    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/src/*.h
+    ${PROJECT_SOURCE_DIR}/3rdParty/qxt/*.h
+    ${PROJECT_SOURCE_DIR}/src/*.h
+)
+
+set (MACX_PROJECT_SOURCES
+    ${PROJECT_SOURCE_DIR}/3rdParty/qxt/qxtglobalshortcut_mac.cpp
+    ${PROJECT_SOURCE_DIR}/src/framelesswindow.mm
+)
+
+set (WIN32_PROJECT_SOURCES
+    ${PROJECT_SOURCE_DIR}/3rdParty/qxt/qxtglobalshortcut_win.cpp
+    ${PROJECT_SOURCE_DIR}/src/framelesswindow.cpp
+)
+
+set (LINUX_PROJECT_SOURCES
+    ${PROJECT_SOURCE_DIR}/3rdParty/qxt/qxtglobalshortcut_x11.cpp
+)
+
+
+add_executable(${PROJECT_NAME} WIN32 MACOSX_BUNDLE ${GENERAL_PROJECT_SOURCES} ${HEADERS})
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+    ${PROJECT_SOURCE_DIR}/src/
+    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/include
+    ${PROJECT_SOURCE_DIR}/3rdParty/qautostart/src
+    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit
+    ${PROJECT_SOURCE_DIR}/3rdParty/qxt
+)
+
+target_compile_definitions(${PROJECT_NAME} PUBLIC
+    APP_VERSION="2.0.0"
```

```suggestion
    APP_VERSION="${CMAKE_PROJECT_VERSION}"
```

Just so we don't have to update the version in two different places.